src/ring_theory/ideal/nc_jacobson_radical.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2022 Haruhisa Enomoto. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Haruhisa Enomoto
+-/
+import ring_theory.jacobson_ideal
+import ring_theory.inverses
+/-!
+# Noncommutative Jacobson radical
+
+In this file, we define the Jacobson radical of a ring and prove its basic properties.
+
+Let `R` be a (possibly noncommutative) ring. The *Jacobson radical* of `R` is defined
+to be the intersection of all maximal *left* ideals of `R`, which coincides with that of
+all maximal *right* ideals, so it is a two-sided ideal.
+
+## Main definitions
+
+* `nc_jacobson R : ideal R`: the Jacobson radical of a ring `R`, that is,
+  the intersection of all maximal *left* ideals.
+
+## Main statements
+
+* `mem_nc_jacobson_tfae`: some equivalent conditions for an element in a ring
+  to be contained in the Jacobson radical.
+
+* `nc_jacobson_op`: the Jacobson radical of a ring coincides with that of its opposite ring.
+  This essentially says that the definition of the Jacobson radical is left-right symmetric:
+  the intersection of all maximal left ideals coincides with that of all maximal right ideals.
```

Comment:
These should all have an `ideal.` prefix, as that way doc-gen will turn them into links.

Reply:
Thanks, I didn't know that. I added a prefix (now it's `ring.` since I've changed namespace).